tests: pipe: Enhance tests to improve code coverage

ajaykish opened the pull request:
This test is intended to validate k_pipe_alloc_init()
and k_pipe_cleanup(), when CONFIG_USERSPACE is not defined.
Also added test to validate pending reader and pending writer
feature in pipe.
Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>

ajaykish commented:
@andrewboie @andyross 
Getting segmentaion fault with test test_pipe_block_writer_wait().
In this test we are creating a situation, where our writer thread is going
into pending state during writing into the pipe. 

During unpending the writer thread (_unpend_thread(thread) call in kernel/pipe.c),
I am getting segmentation fault. Could you please look into this issue.
Or shall I create GH issue for the same ?

dcpleung commented:
@ajaykish The test_pipe_block_writer_wait() issue should have been fixed by #9544.

ajaykish committed:
tests: pipe: Enhance tests to improve code coverage

This test is intended to validate k_pipe_alloc_init()
and k_pipe_cleanup(), when CONFIG_USERSPACE is not defined.
Also added test to validate pending reader and pending writer
feature in pipe.

Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>

dcpleung commented:
k_pipe_alloc_init() fails because the test runs out of memory. It is not a problem with the API. Since the additional tests being introduced in this PR need more memory, you have to make the mem_pool bigger too.

ajaykish commented:
@dcpleung Thanks for the comment. I have increased the mem_pool  size in the new patch and the permission issue is also resolved by using the kernel thread instead of user thread during pipe operation.

ajaykish commented:
@nashif I have addressed all the review comments. With this patch code coverage of kernel/pipe.c will improve to 100%. Could you please review it.
